src/services/logger.service.js

```diff
@@ -48,4 +72,41 @@ function getLogger(req) {
   return logger;
 }
 
-export { walletLoggers, initializeWalletLogger, getLogger };
+/**
+ * @param {HathorWallet} wallet
+ * @param {import('winston').Logger} logger
+ */
+function setupWalletStateLogs(wallet, logger) {
+  const times = {
+    connecting: 0,
+    syncing: 0,
+    processing: 0,
+    ready: 0,
+  };
+  wallet.on('state', state => {
+    switch (state) {
+      case HathorWallet.CONNECTING:
+        times.connecting = Date.now();
+        break;
+      case HathorWallet.SYNCING:
+        times.syncing = Date.now();
+        logger.debug(`state_update[syncing]: time to connect ${(times.syncing - times.connecting) / 1000} seconds`);
+        break;
+      case HathorWallet.PROCESSING:
+        times.processing = Date.now();
+        logger.debug(`state_update[processing]: time to sync ${(times.processing - times.syncing) / 1000} seconds`);
+        break;
+      case HathorWallet.READY:
+        times.ready = Date.now();
+        logger.debug(`state_update[ready]: time to process history ${(times.ready - times.processing) / 1000} seconds`);
+        break;
```

Comment:
question: Should we remove the listener on wallet ready? It doesn't seem to be used anymore.

Reply:
It will be used if the connection is restarted due to some instability.
The logs will then reassign the values and log the time to become ready again.

Leaving this on is a feature that will help understand better how the wallets react when starting for the first time and during a restart.